src/js/components/DataTable/DataTable.js

```diff
@@ -53,6 +53,34 @@ const normalizeProp = (prop, context) => {
   return undefined;
 };
 
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+const objectIs = (x, y) => {
+  if (x === y) {
+    return x !== 0 || 1 / x === 1 / y;
+  }
+  return x !== x && y !== y; // eslint-disable-line no-self-compare
+};
+
+const equals = (x, y) =>
+  typeof Object.is === 'function' ? Object.is(x, y) : objectIs(x, y);
+
+const useStateWithDeps = (initialStateFn, deps) => {
```

Comment:
This approach feels a bit too generic-ish. I'm wondering instead if we should create a more specialized hook called something like `useGroupState` which pulls `buildGroupState` inside it and then has an effect that updates the state when `groups` or `groupBy` change.

Reply:
Thank you for your review. I made `useGroupState` hook. Would you review this?